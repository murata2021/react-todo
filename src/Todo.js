
import React from "react"


const todo=({isCompleted,id,todo,completed,deleteTodo})=>{

    const strikeThrough=isCompleted===true?{textDecoration: "line-through"}:{textDecoration: "None"}
    return(
        <>
            <li><span style={strikeThrough} onClick={()=>completed(id)}>{todo}</span> <button onClick={()=>deleteTodo(id)}>‚ùå</button></li> 
        </>
    )
}

export default todo;