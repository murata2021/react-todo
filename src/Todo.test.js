import React from "react";
import { render, screen,fireEvent} from '@testing-library/react';
import Todo from './Todo';

it('renders without crashing', () => {
  render(<Todo />);
});

it("matches snapshot", function() {

    // Object.defineProperty(window, "localStorage", {
    //     value: {
    //       getItem: jest.fn(() => null),
    //       setItem: jest.fn(() => null)
    //     },
    //     writable: true
    //   });
    
    const { asFragment } = render(<Todo />);

    // expect(window.localStorage.getItem).toHaveBeenCalledTimes(1);

    expect(asFragment()).toMatchSnapshot();
  });


  it("runs the delete function on button click", function() {
    const removeMock = jest.fn();
    const { getByText,debug } = render(<Todo deleteTodo={removeMock} />);
    const deleteButton = getByText("‚ùå");
    fireEvent.click(deleteButton);
    expect(removeMock).toHaveBeenCalled();
  });