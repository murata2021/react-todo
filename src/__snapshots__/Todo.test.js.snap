// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li>
    <span
      style="text-decoration: None;"
    />
     
    <button>
      ❌
    </button>
  </li>
</DocumentFragment>
`;

exports[`matches snapshot when deleting 1`] = `
<DocumentFragment>
  <li>
    <span
      style="text-decoration: None;"
    />
     
    <button>
      ❌
    </button>
  </li>
</DocumentFragment>
`;
